hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -1080,9 +1080,8 @@ void Vgen::emit(const vasm_opc& i) {                         \
   } else {                                                   \
     checkSF(i, StatusFlags::NotV);                           \
     if (!flagRequired(i.fl, StatusFlags::C)) {               \
-      /* Perform the shift and set N and Z. */               \
-      a->arm_opc(gpr_w(i.d), gpr_w(i.s1), i.s0.l());         \
-      a->Bic(vixl::zr, gpr_w(i.d), vixl::zr, SetFlags);      \
+      /* Equivalent to shift and test, set N and Z. */       \
+      a->Tst(gpr_w(i.s1), Operand(~(-1LL << (sz - i.s0.l())))); \
```

Comment:
Hi @swalk-cavium , don't the shlli and shlqi require the destination i.d to be left shifted by i.s0 ?